torch/__init__.py

```diff
@@ -780,6 +782,36 @@ def is_warn_always_enabled():
     """
     return _C._get_warnAlways()
 
+def enable_check_sparse_tensor_invariants(enable: builtins.bool = True):
+    r"""Enable or disable sparse tensor invariants check in
+    sparse tensor constructors.
+
+    .. note::
+
+      By default, the sparse tensor invariants check flag is
+      disabled. Use :func:`is_check_sparse_tensor_invariants_enabled`
+      to get the current state of the flag.
+
+    .. note::
+
+      The sparse tensor invariants check flag has global effect in the
+      sense that it is effective to all sparse tensor constructors,
+      both in Python and ATen.
+
+      The flag can be locally overridden by the ``check_invariants``
+      optional argument of the sparse tensor constructor functions.
+
+    Args:
+        enable (:class:`bool`): If True, enable invariants checks,
+                                otherwise, disable the checks.
+    """
+    torch._C._set_check_sparse_tensor_invariants(enable)
```

Comment:
It's a bit of a nit, but why use `_set` for the internal version and then `enable` for the public?

Also, maybe we can consider only using a Python context manager as the public API?

So then people can do 
```
with torch.sparse.check_sparse_tensor_invariants():
    run_my_model()
```

It's also easier to move a context manager like this to a block of code where they expect an issue or around an op they wrote themselves or to use it as a decorator for a test.

Reply:
There exists now three ways to enable/disable invariant checks:

1. Using context manager:
   ```python
   with torch.sparse.check_sparse_tensor_invariants(enable=True):
       run_my_model()
   ```
2. Using a procedural approach (when context manager is inconvenient due to the required indent, useful in command prompt, for instance):
   ```python
   prev_is_enabled = torch.sparse.check_sparse_tensor_invariants.is_enabled()
   torch.sparse.check_sparse_tensor_invariants.enable()
   run_my_model()
   if not prev_is_enabled:
       torch.sparse.check_sparse_tensor_invariants.disable()
   ```
3. Via decorator (useful in tests):
   ```python
   @torch.sparse.check_sparse_tensor_invariants(enable=True)
   def run_my_model():
       ...
       
   run_my_model()
   ```

Reply:
And the forth way is to use `check_invariants` kw argument in a sparse tensor constructor call.